compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1446,9 +1446,13 @@ rewriteStableQualified env q@(Qualified pkgRef modName obj) =
             PRImport pkgId ->
                 -- TODO We probably want to replace this my a more efficient lookup at some point
                 fmap fst $ find (\(_, dalfPkg) -> dalfPackageId dalfPkg == pkgId) $ MS.toList $ envPkgMap env
-    in case (\unitId -> MS.lookup (unitId, modName) (envStablePackages env)) =<< mbUnitId of
+    in case (\unitId -> MS.lookup (stripStdlibVersion unitId, modName) (envStablePackages env)) =<< mbUnitId of
```

Comment:
Why does `rewriteStableQualified` have to handle the case of the version-id being present?

Reply:
The unit id of the actual stdlib is `daml-stdlib-$SDK_VERSION`. The stable package is stored at key `("daml-stdlib", modName)` so when we do a lookup to figure out if we have a module that we need to translate, we need to strip the version. The alternative would be that we store it at key `("daml-stdlib-$SDK-VERSION", modName)` (which is what we did before this PR) and which removes the need to strip  the version here but then we have to strip the version when writing the LF package and we have to keep the code that adds the version when building up the map.

So the solution in this PR only requires stripping the version in one place instead of having to add it in one place only to strip it in another place.

Reply:
ended up doing it the other way around after all since I forgot that `discardInternalModules` also needs changing so either way, we need to change two places.